pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -789,6 +788,39 @@ func (r *ReconcileBareMetalHost) actionManageSteadyState(prov provisioner.Provis
 	return r.manageHostPower(prov, info)
 }
 
+// A host reaching this action handler should be ready -- a state that
+// it will stay in until the user takes further action. It has been
+// registered with the provisioner once, so we use
+// ValidateManagementAccess() to ensure that is still true. We don't
+// use Adopt() because we don't want Ironic to treat the host as
+// having been provisioned. Then we monitor its power status.
+func (r *ReconcileBareMetalHost) actionManageReady(prov provisioner.Provisioner, info *reconcileInfo) (result reconcile.Result, err error) {
+
+	// We always pass false for credentialsChanged because if they had
+	// changed we would have ended up in actionRegister() instead of
+	// here.
+	provResult, err := prov.ValidateManagementAccess(false)
```

Comment:
Question: will we eventually want to do this in the provisioning/deprovisioning state also to handle an Ironic reschedule in that state?

Reply:
We will need to handle that case, but the way we handle it will be different. If we find that we've re-discovered a node during the middle of a provisioning operation, for example, we will need to restart that operation. The agent relies on the conductor to tell it what to do, so provisioning is going to be broken at that point. So we'll need to power off the host, and start provisioning from scratch. That's going to require some more extensive changes (ValidationManagementAccess will need to tell us somehow that it had to register the host, for example). Since they're more complex, I want to reason through those changes separately.

Reply:
Yeah, not suggesting we tackle it as part of this.
The provisioner's source of truth is Ironic, so if it sees that provisioning hasn't started yet according to the new Ironic it should just start again from scratch. I'd have assumed that Ironic will effect whatever power transitions are necessary to get a host in an unknown initial state to start provisioning, except in the case of adopt.

Reply:
The provisioner within the operator will start over, but IPA will be in the middle of an operation and fail the next time it asks for more instructions. I'm not sure if ironic will correctly deduce that the host needs to be restarted if it can talk to an agent there. We'll have to test it and see what happens.